src/PostgREST/Plan.hs

```diff
@@ -123,7 +125,7 @@ treeRestrictRange _ (ActionMutate _) request = Right request
 treeRestrictRange maxRows _ request = pure $ nodeRestrictRange maxRows <$> request
   where
     nodeRestrictRange :: Maybe Integer -> ReadPlan -> ReadPlan
-    nodeRestrictRange m q@ReadPlan{range_=r} = q{range_=restrictRange m r }
+    nodeRestrictRange m q@ReadPlan{range_=r} = q{range_= if hasLimitZero r then limitZeroRange else restrictRange m r }
```

Comment:
Hm, I see the same logic on ApiRequest
https://github.com/PostgREST/postgrest/blob/c8e4f38984d9194d939bf8b3c31010974f560be4/src/PostgREST/ApiRequest.hs#L357

Could this be wrapped inside a function for the Ranges module?

Reply:
Also maybe a comment in the function on why is it needed